code/game/gamemodes/dynamic/dynamic.dm

```diff
@@ -585,10 +587,10 @@ GLOBAL_VAR_INIT(dynamic_forced_threat_level, -1)
 /// Mainly here to facilitate delayed rulesets. All midround/latejoin rulesets are executed with a timered callback to this proc.
 /datum/game_mode/dynamic/proc/execute_midround_latejoin_rule(sent_rule)
 	var/datum/dynamic_ruleset/rule = sent_rule
+	spend_threat(rule.cost)
+	threat_log += "[worldtime2text()]: [rule.ruletype] [rule.name] spent [rule.cost]"
```

Comment:
If the rule fails to execute this will still spend the threat.

Reply:
It runs clean_up (with built in refunding) on line 607, which makes for better logging and threat management overall. If someone does something stupid later like make the execute delayed, this is also a little more robust in the sense that noone else can spend our threat. Midrounds already relied on spending happening before the final execute which might fail due to no candidates, so this was the intended flow from the start.